doc/userguide/capture-hardware/ebpf-xdp.rst

```diff
@@ -0,0 +1,318 @@
+    defrag: yes
+    # eBPF file containing a 'loadbalancer' function that will be inserted into the
+    # kernel and used as load balancing function
+    #ebpf-lb-file:  /etc/suricata/ebpf/lb.bpf
+    # eBPF file containing a 'filter' function that will be inserted into the
+    # kernel and used as packet filter function
+    # eBPF file containing a 'xdp' function that will be inserted into the
+    # kernel and used as XDP packet filter function
+    ebpf-filter-file:  /etc/suricata/ebpf/bypass_filter.bpf
+    bypass: yes
+    use-mmap: yes
+    ring-size: 200000
+
+
+Setup eBPF load balancing
+-------------------------
+
+Copy the resulting ebpf fiter as needed ::
+
+ cp src/ebpf/lb.bpf /etc/suricata/
+
+We will use ``cluster_ebpf`` in the interface section of af-packet ::
+
+  - interface: eth3
+    threads: 16
+    cluster-id: 97
+    cluster-type: cluster_ebpf
+    defrag: yes
+    # eBPF file containing a 'loadbalancer' function that will be inserted into the
+    # kernel and used as load balancing function
+    ebpf-lb-file:  /etc/suricata/ebpf/lb.bpf
+    use-mmap: yes
+    ring-size: 200000
+
+Setup XDP
+---------
+
+Copy the resulting xdp fiter as needed::
+
+ cp src/ebpf/xdp_filter.bpf /etc/suricata/ebpf/
+
+Setup af-packet section/interface in ``suricata.yaml``.
+
+We will use ``cluster_qm`` as we have symmetric hashing on the NIC, ``xdp-mode: driver`` and we will
+also use the ``/etc/suricata/ebpf/xdp_filter.bpf`` (in our example TCP offloading/bypass) ::
+
+  - interface: eth3
+    threads: 16
+    cluster-id: 97
+    cluster-type: cluster_qm # symmetric hashing is a must!
+    defrag: yes
+    # eBPF file containing a 'loadbalancer' function that will be inserted into the
+    # kernel and used as load balancing function
+    #ebpf-lb-file:  /etc/suricata/ebpf/lb.bpf
+    # eBPF file containing a 'filter' function that will be inserted into the
+    # kernel and used as packet filter function
+    # eBPF file containing a 'xdp' function that will be inserted into the
+    # kernel and used as XDP packet filter function
+    #ebpf-filter-file:  /etc/suricata/ebpf/filter.bpf
+    # Xdp mode, "soft" for skb based version, "driver" for network card based
+    # and "hw" for card supporting eBPF.
+    xdp-mode: driver
+    xdp-filter-file:  /etc/suricata/ebpf/xdp_filter.bpf
+    # if the ebpf filter implements a bypass function, you can set 'bypass' to
+    # yes and benefit from these feature
+    bypass: yes
+    use-mmap: yes
+    ring-size: 200000
+
+
+XDP bypass is compatible with AF_PACKET IPS mode. Packets from bypassed flows will be send directly 
+from one card to the second card without going by the kernel network stack.
+
+Setup symmetric hashing on the NIC
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Follow these instructions closely for desired result::
+
+ ifconfig eth3 down
+
+Use in tree kernel drivers, XDP support is not available in out of source Intel drivers.
```

Comment:
*out of source* drivers? You mean those not within the kerneltree?

Reply:
Fixed: Intel website driver don't have the feature